src/AMSlib/AMS.cpp

```diff
@@ -38,92 +408,130 @@ void _AMSExecute(AMSExecutor executor,
                  const void **input_data,
                  void **output_data,
                  int inputDim,
-                 int outputDim,
-                 MPI_Comm Comm = 0)
+                 int outputDim)
 {
-  uint64_t index = reinterpret_cast<uint64_t>(executor);
+  long index = static_cast<long>(executor);
```

Comment:
is long (could be 32-bit)  intentional instead of the fixed bitwidth uint64_t?

Reply:
Added a int64_t. Fixed.